replica/replica.go

```diff
@@ -242,20 +248,27 @@ func (replica *Replica) filterHeight(height process.Height) bool {
 	return height >= replica.proc.CurrentHeight
 }
 
-func (replica *Replica) filterFrom(from id.Signatory) bool {
-	return replica.procsAllowed[from]
-}
-
 func (replica *Replica) flush() {
+	copied := map[id.Signatory]bool{}
+	for sig := range replica.procsAllowed {
+		copied[sig] = true
```

Comment:
Why do we copy the map? I might be missing something, but it seems unnecessary (and inefficient, since we have to do it O(n) times for n messages).

Reply:
You're right, I thought there is a race condition on `replica.procsAllowed`, but seems it's not the case.
Remove the map copying